airflow/config_templates/airflow_local_settings.py

```diff
@@ -286,7 +286,7 @@
             "task": {
                 "class": "airflow.providers.google.cloud.log.stackdriver_task_handler.StackdriverTaskHandler",
                 "formatter": "airflow",
-                "name": log_name,
+                "gcp_log_name": log_name,
```

Comment:
How is this backward compatible?
What happens if user has latest airflow version with older version of Google provider?

Reply:
Ah sorry forgot discussion in https://github.com/apache/airflow/pull/38071#discussion_r1529196456
I think the newsgragment should be updated to reflect why we are going this. It should reflect more than just what is changed.

Reply:
We can keep both attributes for the moment (but we will have a warning message at instanciation).

Reply:
If there are no valid use cases for current behavior then there is no reason to keep it. It may cause more confusion.

But others might have different view on this
cc @potiuk @Taragolis @hussein-awala